internal/compiler/passes/materialize_fake_properties.rs

```diff
@@ -93,7 +93,12 @@ fn should_materialize(
     }
     let has_declared_property = match base_type {
         ElementType::Component(c) => has_declared_property(&c.root_element.borrow(), prop),
-        ElementType::Builtin(b) => b.properties.contains_key(prop),
+        ElementType::Builtin(b) => {
+            if let Some(info) = b.reserved_properties.get(prop) {
+                return Some(info.ty.clone());
+            }
+            b.properties.contains_key(prop) || b.reserved_properties.contains_key(prop)
```

Comment:
the second part of the `||` can never be true, right? (since we returned a couple of lines before)

Reply:
Indeed :)